fix: release please configs

Stromweld opened the pull request:
## Description

This PR modernizes the kitchen-docker project across four areas: release automation, CI pipeline, integration test tooling, and Windows container support.

### Release Please

- Adds `release-please-config.json` and `.release-please-manifest.json` for config-file-based release-please v5 (replaces deprecated inline `release-type`/`package-name`/`version-file` params in `publish.yaml`)

### CI Pipeline (`ci.yml` → `lint.yml`)

- Renamed workflow to better reflect scope: lint, unit, and integration tests
- Added `CHEF_LICENSE_KEY` secret injection for licensed tooling
- Removed `docker/setup-buildx-action` step from Windows job — the `moby/buildkit:buildx-stable-1` image has no Windows manifest and the GHA `windows-2022` runner already ships buildx as a CLI plugin
- Updated capabilities matrix: replaced EOL `ubuntu-2004` with `ubuntu-2604`

### Integration Test Tooling

- Replaced `kitchen-inspec` + embedded `lib/train/docker.rb` with `kitchen-cinc-auditor` + `cinc-auditor-bin` (open-source Cinc equivalents)
- Removed `lib/train/docker.rb` (embedded train transport, 126 lines) — uses upstream `train-docker` gem instead
- Removed `lib/docker/version.rb` (superseded by `lib/kitchen/docker/docker_version.rb`)
- Added `test/cookbooks/cinc_test` and `test/cookbooks/docker_test` cookbooks for integration test suites
- Added cross-platform InSpec profile (`test/integration/cinc/inspec/cinc_spec.rb`)
- Updated `Gemfile` and `Rakefile` accordingly; switched `.rubocop.yml` from `chefstyle` to `cookstyle/chefstyle`

### Windows Container Support

- Fixed Docker socket detection: replaced `windows_os?` (which checked the *target platform name*) with `Gem.win_platform?` (which checks the *host OS*) in both `driver.rb` and `transport.rb`. Without this fix, Kitchen used `unix:///var/run/docker.sock` on a Windows host instead of `npipe:////./pipe/docker_engine`.
- Added `windows-2022` platform to `kitchen.yml` using `mcr.microsoft.com/windows/servercore:ltsc2022` — a true Windows container, since GHA `windows-2022` runners run Docker EE in Windows container mode and cannot pull Linux images
- Installs Cinc via `provision_command` PowerShell one-liner; sets `cinc_omnibus_root` so Kitchen resolves the correct client path (`C:\cinc-project\cinc\bin\cinc-client.bat`)
- Uses `verifier: name: dummy` for the Windows platform — `cinc_auditor` uses WinRM for Windows targets but Windows Server Core containers do not have WinRM configured; verify support is a follow-up

## Issues Resolved

N/A — proactive modernization and CI fix.

## Type of Change

`fix` — corrects broken Windows CI and updates tooling/release automation.

## Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] Commit message includes a [Conventional Commit Message](https://www.conventionalcommits.org/en/v1.0.0)

---

> _This PR was created with AI assistance using [Hermes Agent](https://hermes-agent.nousresearch.com) v0.17.0 (claude-sonnet-4.6)._

Stromweld committed:
fix: release please configs

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
add codeowners

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

damacus reviewed (commented):
Where are we at with this?

damacus commented:
Blocks: #421 #422 #423 #424 #425 #427 
cc/ @RulerOf

Stromweld committed:
Merge remote-tracking branch 'origin/main' into fix-release-please

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
updates

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld commented:
sorry this fell off my radar

Stromweld committed:
fix cookstyle

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
update rubocop to use cookstyle

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
linting fixes

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test cinc changes

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
fix load errors

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
cleanup

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
fix: remove Docker Buildx step from Windows CI job

The moby/buildkit:buildx-stable-1 image has no Windows manifest, causing
the docker-container driver to fail with "no matching manifest for
windows(10.0.26100)/amd64". The Windows runner already has buildx
installed as a CLI plugin, so the setup-buildx-action is unnecessary.

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
fix: use host OS detection for Docker socket configuration

The windows_os? method checks the target platform name (e.g. ubuntu-24.04),
not the host machine running Kitchen. When running on a Windows host targeting
Linux containers, this caused the socket to incorrectly default to
unix:///var/run/docker.sock instead of npipe:////./pipe/docker_engine.

Replace driver.windows_os? / transport.windows_os? with Gem.win_platform?
which correctly detects the actual host operating system.

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
fix: use Windows containers for Windows CI integration test

The Windows CI job was attempting to run Linux containers (ubuntu-24.04)
on a Windows Docker daemon in Windows container mode, which is impossible.
The Docker daemon on GHA windows-latest cannot pull Linux images.

Switch the Windows integration test to use actual Windows containers:
- Add windows-2022 platform using mcr.microsoft.com/windows/servercore:ltsc2022
- Update workflow matrix to target windows-2022 instead of ubuntu-2404
- Remove DOCKER_DEFAULT_PLATFORM env (not needed for native Windows containers)
- Make InSpec cinc-client version check cross-platform

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
fix: set cinc_omnibus_root for Windows container platform

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
tes

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
fix: skip InSpec verify for Windows container (cinc_auditor uses WinRM, not docker exec)

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
fix: remove install_strategy override for Windows platform (inherit global skip)

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

Stromweld committed:
fix spec location

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

damacus commented on .github/workflows/lint.yml:
```diff
@@ -1,9 +1,12 @@
 "on":
   pull_request:
 
+env:
```
We don't need this if we're using cinc stuff

Stromweld replied:
I ran into issues where cinc_auditor was still installing inspec and running inspec under the hood. This caused the license error on verify phase. We need to do a deeper dive and verify kitchen-cinc auditor is using cinc-auditor and there are no dependencies on inspec or inspec-core.

Stromweld committed:
test

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>
